src/main/java/org/apache/ibatis/migration/hook/Hook.java

```diff
@@ -0,0 +1,27 @@
+/**
+ *    Copyright 2010-2017 the original author or authors.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package org.apache.ibatis.migration.hook;
+
+import java.util.Map;
+
+/**
+ * @author cbongiorno on 12/28/17.
+ */
+public interface Hook {
+  void before(Map<String, Object> bindingMap);
```

Comment:
These should return a `<T> ` so that the results of the hook can be used (or not). The use of the `Hook` class was kind-of a necessity because I no longer had a `FileMigrationHook` I had a `NewHook` But creating a `NewHook` seemed silly because it's just a dumb implementation of any hook. Hence the `SimpleHook` because I didn't want to have a command-specific hook when the class can be conceptually reused.